core/src/main/java/org/bitcoinj/wallet/WalletProtobufSerializer.java

```diff
@@ -753,7 +757,6 @@ private void readConfidence(final NetworkParameters params, final Transaction tx
                 log.warn("Have depth but not BUILDING for tx {}", tx.getHashAsString());
                 return;
             }
-            confidence.setDepthInBlocks(confidenceProto.getDepth());
```

Comment:
Changing the protobuf? Is this what is causing the many newly broken tests?

Reply:
Damn "newly broken tests". Tests fail very randomly. I don't know yet under which conditions I only get 12 failing tests but so far I only and often observed it on our multi-hd-accounts branch.

Reply:
I suppose that using context which is thread specific for passing number of blocks is not a good idea...
Still thinking how to fix this.

Reply:
Just tested multiple_hd_accounts, received 38 tests failed...